easybuild/easyconfigs/t/TomoPy/TomoPy-1.11.0-foss-2020b.eb

```diff
@@ -0,0 +1,53 @@
+# Authors: Joachim Hein, Zdenek Matej, Lund University
+
+easyblock = 'PythonPackage'
+
+name = 'TomoPy'
+version = '1.11.0'
+
+github_account = 'tomopy'
+homepage = "https://tomopy.readthedocs.io/en/latest/"
+description = """
+Tomographic reconstruction creates three-dimensional views of an object by 
+combining two-dimensional images taken from multiple directions, for example, 
+this is how a CAT (computer-aided tomography) scanner generates 3D 
+views of the heart or brain.
+"""
+
+toolchain = {'name': 'foss', 'version': '2020b'}
+
+dependencies = [
+    ('Python', '3.8.6'),
+    ('SciPy-bundle', '2020.11'),
+    ('scikit-image', '0.18.1'),
+    ('OpenCV', '4.5.1', '-contrib'),
+    ('imkl', '2020.4.304'),
+    ('numexpr', '2.8.1'),
+    ('DXchange', '0.1.8'),
+    ('PyWavelets', '1.1.1'),
+]
+
+builddependencies = [
+    ('scikit-build', '0.11.1'),
```

Comment:
I suspect this isn't actually used at all? https://github.com/tomopy/tomopy/blob/master/CMakeLists.txt#L12 since no such parameter is specified? Isn't it just a normal cmake project?

Reply:
Hi @Micket , I think, based on a personal discussion with the scientific developer long time ago, it is opposite and they are heavily relying on `scikit-build` and are actually managing the build process with it. It is again based on TomoPy [azure-pipelines.yml](https://github.com/tomopy/tomopy/blob/1.11.0/azure-pipelines.yml).

Reply:
To test, I removed the scikit-build from the config and rebuild TomoPy.  So the dependency section looks like:

```
builddependencies = [
    ('CMake', '3.18.4'),
    ('Ninja', '1.10.1'),
]


```

and I get the error as below:

```
    ERROR: Command errored out with exit status 1:
     command: /sw/easybuild/software/Python/3.8.6-GCCcore-10.2.0/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/eb-fpX2V6/pip-req-build-x4mbpfab/\
setup.py'"'"'; __file__='"'"'/tmp/eb-fpX2V6/pip-req-build-x4mbpfab/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', \
'"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/eb-fpX2V6/pip-pip-egg-info-dxvpkp5o
         cwd: /tmp/eb-fpX2V6/pip-req-build-x4mbpfab/
    Complete output (5 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/eb-fpX2V6/pip-req-build-x4mbpfab/setup.py", line 9, in <module>
        from skbuild import setup
    ModuleNotFoundError: No module named 'skbuild'
```

So it is needed in our current set up.

Reply:
https://tomopy.readthedocs.io/en/latest/devguide.html#building-tomopy